src/tools/perf/lib/libperf.c

```diff
@@ -1413,9 +1457,15 @@ static ucs_status_t uct_perf_setup(ucx_perf_context_t *perf)
     status = uct_perf_test_check_capabilities(params, perf->uct.iface,
                                               perf->uct.md);
     /* sync status across all processes */
-    status = ucp_perf_test_exchange_status(perf, status);
-    if (status != UCS_OK) {
-        goto out_iface_close;
+    for (i = 0; i < group_size; ++i) {
```

Comment:
why need this loop?

Reply:
UCT (before mine UCT patch) works this way - I have to be homogeneous in UCT case before my UCT patch

Reply:
but UCT will also not work with group_size>2 right?
if we keep this loop, what is the point of moving it from exchange_status into here?

Reply:
I didn't find such assumption in UCT case, UCT always supposes any group_size

Reply:
exchange_status used by UCP as well